src/PostgREST/App.hs

```diff
@@ -257,8 +269,16 @@ app dbStructure conf apiRequest =
 
       mapSnd f (a, b) = (a, f b)
       readReq = readRequest (configMaxRows conf) (dbRelations dbStructure) (map (mapSnd pdReturnType) $ dbProcs dbStructure) apiRequest
+      fldNames = fieldNames <$> readReq
+      fieldNames :: ReadRequest -> [FieldName]
```

Comment:
This feels out of place here somehow. The surrounding statements are making helper variables and this is a full function. Maybe put it lower in the file or in another module?